fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -2309,6 +2354,68 @@ void MultiVersionDatabase::DatabaseState::close() {
 	});
 }
 
+namespace {
+
+const char* inializationStateToString(MultiVersionDatabase::InitializationState initState) {
+	switch (initState) {
+	case MultiVersionDatabase::InitializationState::INITIALIZING:
+		return "initializing";
+	case MultiVersionDatabase::InitializationState::INITIALIZATION_FAILED:
+		return "initialization_failed";
+	case MultiVersionDatabase::InitializationState::CREATED:
+		return "created";
+	case MultiVersionDatabase::InitializationState::INCOMPATIBLE:
+		return "incompatible";
+	case MultiVersionDatabase::InitializationState::CLOSED:
+		return "closed";
+	default:
+		ASSERT(false);
+		return "invalid_state";
+	}
+}
+
+} // namespace
+
+Standalone<StringRef> MultiVersionDatabase::DatabaseState::getClientStatus(
```

Comment:
As a reviewer, it was quite confusing to have this *internal* function to be *also* named `getClientStatus()` along with other public interface functions.
Also it seems worthwhile to mention via comment that it combines the `ClientInfo` taken from `DatabaseContext` with MVC's client information.

+Why are we serialize-deserializing JSON from `DatabaseContext::getClientStatus()` instead of directly passing `json_spirit::mValue` if it's passed from main thread to main thread?

Reply:
If complexity of moving around nested `std::map` is a concern, you can pass around `Reference<json_spirit::mValue>` instead.

Reply:
To access the DatabaseContext MVC goes over the C API of an external client. We don't pass C++ objects over C API for compatibility reasons. The data passed must be either a stable C structure defined in fdb_c.h or it must be passed serialized. In this case we use JSON for serialization.

Reply:
I'm not sure I understand. `Standalone<StringRef>` is also a C++ object.

Reply:
It is just for managing memory of StringRef, which is built from the string pointer coming from the API. 
Look at the implementation of DLDatabase::getClientStatus and fdb_database_get_client_status.
fdb_database_get_client_status returns a future, then we have fdb_future_get_key to extract string from the future (it is a bit unlucky name, because it is used not only for keys but for strings in general). Note that fdb_database_get_client_status can be implemented either by MultiVersionDatabase::getClientStatus or ThreadSafeDatabase::getClientStatus, depending on the client configuration. Both of them now return serialized JSON strings.

Reply:
I don't expect this to be performance critical, but if it becomes critical, then should use something more efficient than JSON in general, but at the moment it is consistent with the server status which is also reported as JSON.

Reply:
I added a comment to DatabaseState::getClientStatus